execution_engine/src/core/engine_state/mod.rs

```diff
@@ -1688,7 +1688,20 @@ where
         Error: From<S::Error>,
     {
         self.state
-            .missing_trie_keys(correlation_id, trie_key)
+            .missing_trie_keys(correlation_id, vec![trie_key])
+            .map_err(Error::from)
+    }
+
+    pub fn trie_integrity_check(
+        &self,
+        correlation_id: CorrelationId,
+        trie_keys: Vec<Blake2bHash>,
+    ) -> Result<Vec<Blake2bHash>, Error>
+    where
+        Error: From<S::Error>,
+    {
+        self.state
+            .missing_trie_keys(correlation_id, trie_keys)
```

Comment:
Can't we merge those two functions into one that accepts the vector of hashes?

Reply:
We could, but I thought it made sense to keep those functions as distinct as possible. I thought it made more sense to split off to a once off function that is named as such and used for that once specific reason.

Reply:
Depends on how many places the single-hash version is called. If it's only one, and we don't break call sides, I would merge it. Not going to push, just a preference.